Lib/email/_header_value_parser.py

```diff
@@ -1513,16 +1513,22 @@ def get_obs_local_part(value):
                 raise
             token, value = get_cfws(value)
         obs_local_part.append(token)
-    if (obs_local_part[0].token_type == 'dot' or
-            obs_local_part[0].token_type=='cfws' and
-            obs_local_part[1].token_type=='dot'):
+    if not obs_local_part:
         obs_local_part.defects.append(errors.InvalidHeaderDefect(
-            "Invalid leading '.' in local part"))
-    if (obs_local_part[-1].token_type == 'dot' or
-            obs_local_part[-1].token_type=='cfws' and
-            obs_local_part[-2].token_type=='dot'):
-        obs_local_part.defects.append(errors.InvalidHeaderDefect(
-            "Invalid trailing '.' in local part"))
+            "abandoned parse; truncated value?"))
+    else:
+        if (obs_local_part[0].token_type == 'dot'
+            or (obs_local_part[0].token_type=='cfws'
+                and len(obs_local_part) == 1
+                or obs_local_part[1].token_type=='dot')):
```

Comment:
Please add tests for all these new cases. For example this condition is passed if there are more that 1 parts and the second part is a dot. Does it report a defect "Invalid leading '.' in local part" for "python.org"?